react/src/views/BoxEdit/components/BoxEdit.tsx

```diff
@@ -127,6 +129,16 @@ function BoxEdit({
     tags: boxData?.tags || [],
   };
 
+  if (boxData?.product?.deletedOn !== null) {
+    defaultValues.productId = { label: "", value: "" };
```

Comment:
@serchtul This should probably be null or undefined to flush the value in the product field properly. I can imagine that this will cause a type error based on the type definition through zod. Try resolving this type error and we can maybe talk about it tomorrow.

Reply:
in line 65 you have more information on the problem there.